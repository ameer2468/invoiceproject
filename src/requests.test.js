import axios from "axios";

const api = process.env.NEXT_PUBLIC_API_URL;
const apiKey = process.env.NEXT_PUBLIC_API_KEY;
test("get invoices", async () => {
  await axios
    .get(`${api}/invoices`, {
      headers: {
        "x-api-key": apiKey,
      },
      params: {
        from: "Ameer",
      },
    })
    .then((res) => {
      expect(res.status).toBe(200);
      expect(res.data[0]).toEqual(
        expect.objectContaining({
          from: "Ameer",
        })
      );
    });
});

test("get an invoice", async () => {
  await axios
    .get(`${api}/invoice`, {
      headers: {
        "x-api-key": apiKey,
      },
      params: {
        id: "324345756",
      },
    })
    .then((res) => {
      expect(res.status).toBe(200);
      expect(res.data.invoice[0]).toEqual(
        expect.objectContaining({
          id: "324345756",
        })
      );
    });
});
