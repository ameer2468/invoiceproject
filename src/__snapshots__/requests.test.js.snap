// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get invoices 1`] = `
Object {
  "amount": "2323.00",
  "date": "2022-06-30T00:00:00.000Z",
  "description": "asdasdasdasdsada",
  "from": "Ameer",
  "id": "324345756",
  "status": "paid      ",
  "to": "sdfsd",
}
`;
